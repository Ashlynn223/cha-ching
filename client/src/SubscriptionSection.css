@font-face {
  font-family: "Inter-Thin";
  src: url("Inter-Thin.ttf");
}

@font-face {
  font-family: "Inter-Bold";
  src: url("Inter-Bold.ttf");
}

.subscriptionSectionBackground {
  background-image: url("./goldenflower.jpg");
  background-position: no-repeat;
  background-size: cover;
  display: flex;
  flex-direction: column;
  height: 100vh;
  width: 100vw;
}

.pageContents {
  display: flex;
  justify-content: center;
  flex-direction: column;
  position: relative;
}

.header {
  display: flex;
  font-family: "Inter-Thin";
  font-size: 73px;
  font-style: italic;
  justify-content: center;
  padding: 20px 0 20px 0;
}

.cards {
  display: flex;
  justify-content: center;
}

.button {
  align-items: center;
  background-color: #e4987c;
  border-radius: 0;
  border: none;
  color: white;
  cursor: pointer;
  display: flex;
  font-family: "Inter-Bold";
  font-size: 40px;
  font-style: bold;
  font-weight: 800;
  justify-content: center;
  line-height: 80px;
  text-align: center;
  text-shadow: 1px 1px 2px black;
  width: 100vw;
}

.button:hover {
  background-color: white;
  color: #e4987c;
  text-shadow: 1px 1px 2px white;
}
.footer {
  background-color: black;
  height: 15vh;
  width: 100vw;
}

@media (max-width: 660px) {
  .subscriptionSectionBackground {
    height: 200%;
  }

  .header {
    display: flex;
    font-size: 32px;
    font-weight: 600;
    padding: 20px 0 20px 0;
  }
  .cards {
    flex-direction: column;
    margin: 0 20px 0 20px;
  }

  .button {
    font-size: 20px;
  }
}

