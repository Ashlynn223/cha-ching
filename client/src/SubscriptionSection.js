import "./SubscriptionSection.css";
import ContinueIcon from "./ContinueIcon.js";
import HSpacer from "./HSpacer";
import React, { useState, useEffect } from "react";
import SubscriptionDetailsCard from "./SubscriptionDetailsCard";
import { useNavigate } from "react-router-dom";

const color = "white";
const SubscriptionSection = () => {
  const [onHover, setOnHover] = useState(false);
  const [selectedCardId, setSelectedCardId] = useState(false);
  const [subscriptionType, setSubscriptionType] = useState("starter");
  const [subscriptionPriceText, setSubscriptionPriceText] = useState("55");
  const [stripeSubscriptionPrice, setStripeSubscriptionPrice] = useState(5500);
  const [stripePriceId, setStripePriceId] = useState(process.env.REACT_APP_STARTER_STRIPE_PRICE_ID)
  const navigate = useNavigate();

  useEffect(() => {
    switch(subscriptionType) {
      case "starter": 
        setStripePriceId(process.env.REACT_APP_STARTER_STRIPE_PRICE_ID);
        break;
      case "intermediate":
        setStripePriceId(process.env.REACT_APP_INTERMEDIATE_STRIPE_PRICE_ID);
        break;
      case "advanced":
        setStripePriceId(process.env.REACT_APP_ADVANCED_STRIPE_PRICE_ID);
      break;
      default:
        setStripePriceId(process.env.REACT_APP_STARTER_STRIPE_PRICE_ID)
    }
  },[subscriptionType])

  const handleSubmit = () => {
    const requestOptions = {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        priceId: stripePriceId
      }),
    };

    fetch(
      `${process.env.REACT_APP_API_BASE_URL}/create-incomplete-subscription`,
      requestOptions,
    )
      .then((res) => res.json())
      .then((data) => {
        navigate("/checkout", {
          state: {
            clientSecret: data.latest_invoice.payment_intent.client_secret,
            subscriptionType: subscriptionType,
            subscriptionPrice: subscriptionPriceText,
          },
        });
      })
      .catch((err) => console.log("error creating product:", err));
  }

  return (
    <div className="subscriptionSectionBackground">
      <div className="pageContents">
        <div className="header">Select Your Subscription</div>
        <div className="cards">
          <SubscriptionDetailsCard
            id={"starter"}
            selectedCardId={selectedCardId}
            subscriptionName="Starter"
            price={"55"}
            subscriptionDetails={[
              "Group classes with apprentice and intermediate instructors",
              "Weekly instructional videos",
              "Exclusive access to our online community of ikebana enthusiastsr",
              "Bi-weekly flower delivery",
              "Includes initial starter kit (kenzan, vase, scissors)",
              "Small group sessions with other advancing peers"
            ]}
            onClick={() => {
              setSelectedCardId("starter")
              setSubscriptionType("starter")
              setSubscriptionPriceText("55")
              setStripeSubscriptionPrice(5500)
            }}
          />
          <SubscriptionDetailsCard
            id={"intermediate"}
            selectedCardId={selectedCardId}
            subscriptionName="Intermediate"
            price={"115"}
            subscriptionDetails={[
              "Exclusive access to our online community of ikebana enthusiasts",
              "Weekly step-by-step video tutorials and instructional pamphlets",
              "Ikebana maintenance tips",
              "Includes initial starter kit (kenzen, vase, scissors)",
              "Syllabus of recommended readings",
            ]}
            onClick={() => {
              setSelectedCardId("intermediate")
              setSubscriptionType("intermediate")
              setSubscriptionPriceText("115")
              setStripeSubscriptionPrice(11500)
            }}
          />
          <SubscriptionDetailsCard
            id={"advanced"}
            selectedCardId={selectedCardId}
            subscriptionName="Advanced"
            price={"205"}
            subscriptionDetails={[
              "Zoom one-on-one sessions with a Master",
              "Monthly instructional videos, advanced techniques from a Master",
              "Invitations to annual ikebana events and competitions",
              "Weekly flower delivery and Seasonal themed vases",
              "Exclusive access to our online community of ikebana enthusiasts",
              "Includes everything from starter and intermediate"
            ]}
            onClick={() => {
              setSelectedCardId("advanced")
              setSubscriptionType("advanced")
              setSubscriptionPriceText("205")
              setStripeSubscriptionPrice(20500)
            }}
          />
        </div>
        <button
          aria-label="submit a subscription type"
          role="button"
          onMouseEnter={() => setOnHover(true)}
          onMouseLeave={() => setOnHover(false)}
          className="button"
          onClick={() => handleSubmit()}
        >
          Select a Subscription Type
          <HSpacer />
          <ContinueIcon color={onHover ? "#e4987c" : color} />
        </button>
        <div className="footer" />
      </div>
    </div>
  );
};

export default SubscriptionSection;
