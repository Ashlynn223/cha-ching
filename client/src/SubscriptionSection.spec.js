import { render, screen } from "@testing-library/react";
import SubscriptionSection from "./SubscriptionSection.js";

describe("<SubscriptionSection />", () => {
  it("renders correctly", () => {
    const component = render(<SubscriptionSection onClick={() => {}} />);
    expect(component.asFragment()).toMatchSnapshot();
  });

  it("renders a subscription button", () => {
    render(<SubscriptionSection />);
    expect(
      screen.getByRole("button", { name: "submit a subscription type" }),
    ).toBeVisible();
    expect(screen.getByText("Select a Subscription Type")).toBeVisible();
  });
});
