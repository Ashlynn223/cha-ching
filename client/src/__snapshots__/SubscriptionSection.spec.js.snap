// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubscriptionSection /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="subscriptionSectionBackground"
  >
    <div
      class="pageContents"
    >
      <div
        class="header"
      >
        Select Your Subscription
      </div>
      <div
        class="cards"
      />
      <button
        aria-label="submit a subscription type"
        class="button"
        role="button"
      >
        Select a Subscription Type
        <div
          style="min-width: 8px; height: 100%;"
        />
        <svg
          fill="none"
          height="65"
          viewBox="0 0 65 65"
          width="65"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            filter="url(#filter0_d_248_229)"
          >
            <circle
              cx="32.5"
              cy="28.5"
              r="25.5"
              shape-rendering="crispEdges"
              stroke="white"
              stroke-width="6"
              transform="rotate(-90 32.5 28.5)"
            />
          </g>
          <g
            filter="url(#filter1_d_248_229)"
          >
            <path
              clip-rule="evenodd"
              d="M13.0234 28.7375L25.5599 41.6488L25.5602 41.6485L28.9359 45.125L44.8484 28.7375L28.9359 12.35L25.5598 15.8269L25.5594 15.8264L13.0234 28.7375ZM25.5598 15.8269L13.0234 28.7375L25.5602 41.6485L38.0967 28.7379L25.5598 15.8269Z"
              fill="white"
              fill-rule="evenodd"
            />
          </g>
          <defs>
            <filter
              color-interpolation-filters="sRGB"
              filterUnits="userSpaceOnUse"
              height="65"
              id="filter0_d_248_229"
              width="65"
              x="0"
              y="0"
            >
              <feflood
                flood-opacity="0"
                result="BackgroundImageFix"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feoffset
                dy="4"
              />
              <fegaussianblur
                stdDeviation="2"
              />
              <fecomposite
                in2="hardAlpha"
                operator="out"
              />
              <fecolormatrix
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
              />
              <feblend
                in2="BackgroundImageFix"
                mode="normal"
                result="effect1_dropShadow_248_229"
              />
              <feblend
                in="SourceGraphic"
                in2="effect1_dropShadow_248_229"
                mode="normal"
                result="shape"
              />
            </filter>
            <filter
              color-interpolation-filters="sRGB"
              filterUnits="userSpaceOnUse"
              height="40.7749"
              id="filter1_d_248_229"
              width="39.8281"
              x="9.02344"
              y="12.3501"
            >
              <feflood
                flood-opacity="0"
                result="BackgroundImageFix"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feoffset
                dy="4"
              />
              <fegaussianblur
                stdDeviation="2"
              />
              <fecomposite
                in2="hardAlpha"
                operator="out"
              />
              <fecolormatrix
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
              />
              <feblend
                in2="BackgroundImageFix"
                mode="normal"
                result="effect1_dropShadow_248_229"
              />
              <feblend
                in="SourceGraphic"
                in2="effect1_dropShadow_248_229"
                mode="normal"
                result="shape"
              />
            </filter>
          </defs>
        </svg>
      </button>
      <div
        class="footer"
      />
    </div>
  </div>
</DocumentFragment>
`;
